java/arcs/android/storage/database/AndroidSqliteDatabaseManager.kt

```diff
@@ -103,8 +105,15 @@ class AndroidSqliteDatabaseManager(
     override suspend fun removeExpiredEntities(): Job = coroutineScope {
         launch {
             registry.fetchAll()
-                .map { getDatabase(it.name, it.isPersistent) }
-                .forEach { it.removeExpiredEntities() }
+                .map { it.name to getDatabase(it.name, it.isPersistent) }
+                .forEach { (name, db) ->
+                    if (databaseSizeTooLarge(name)) {
+                        // If the database size is too large, we clear it entirely.
+                        db.reset()
```

Comment:
Wouldn't you want to try and removeExpiredEntities first, and if it's still too big afterwards, then wipe it? Eagerly wiping it seems scary to me.

Reply:
One thing we're worried about is a case where the attempt to remove expired entities outlasts the duration we're allowed to perform an operation during a WorkManager job run.